pkg/operator/types.go

```diff
@@ -53,3 +53,25 @@ func MakeHostAliases(input []monitoringv1.HostAlias) []v1.HostAlias {
 
 	return output
 }
+
+func MakeK8sTopologySpreadConstraint(podLabels map[string]string, tsc monitoringv1.TopologySpreadConstraint) v1.TopologySpreadConstraint {
+	if len(tsc.AdditionalLabelSelectors) > 0 {
+		if tsc.LabelSelector == nil {
+			tsc.LabelSelector = &metav1.LabelSelector{
+				MatchLabels: make(map[string]string),
+			}
+		}
+
+		labelMap := map[monitoringv1.AdditionalLabelSelector]string{
+			"ShardName":    "operator.prometheus.io/shard",
+			"ResourceName": "operator.prometheus.io/name",
+		}
+
+		for _, label := range tsc.AdditionalLabelSelectors {
+			labelName := labelMap[label]
```

Comment:
As a precaution, we need to return an error if the label doesn't match an entry in the map.

Reply:
Approach is different now.